docs/en/development/data_structures.md

```diff
@@ -5,4 +5,879 @@ slug: /internals
+
+	Parent    Ino  // inode of parent; 0 means tracked by parentKey (for hardlinks)
+	Full      bool // the attributes are completed or not
+	KeepCache bool // whether to keep the cached page or not
+}
+```
+
+A few of the fields that need clarification.
+
+- Atime/Atimensec: set only when the file is created and when `SetAttr` is actively called, while accessing and modifying the file normally does not affect the Atime value
+- Nlink
+  - Directory file: initial value is 2 ('.' and '..'), add 1 for each subdirectory
+  - Other files: initial value is 1, add 1 for each hard link created
+- Length
+  - Directory file: fixed at 4096
+  - Soft link (symbolic link) file: length of the string of the path to which the link points
+  - Other files: the length of the actual content of the file
+
+This structure is usually encoded in binary format and stored in the metadata engine.
+
+#### 3.1.6 Edges
+
+Records information about each edge in the file tree, as follows
+
+```
+parentInode, name -> type, inode
+```
+
+where parentInode is the inode number of the parent directory, and the others are the name, type, and inode number of the child files, respectively.
+
+#### 3.1.7 LinkParent
+
+Records the parent directory of some files. The parent directory of most files is recorded in the Parent field of the attribute; however, for files that have been created with hard links, there may be more than one parent directory, so the Parent field is set to 0, and all parent inodes are recorded independently, as follows
+
+```
+inode -> parentInode, links
+```
+
+where links is the count of the parentInode, because multiple hard links can be created in the same directory, and these hard links share the inode.
+
+#### 3.1.8 Chunk
+
+Records information about each Chunk, as follows
+
+```
+inode, index -> []Slices
+```
+
+where inode is the inode number of the file to which the Chunk belongs, and index is the number of all Chunks in the file, starting from 0. The Chunk value is an array of Slices. Each Slice represents a piece of data written by the client, and is appended to this array in the order of writing time. When there is an overlap between different Slices, the later Slice is used.
+
+```go
+type Slice struct {
+	Pos  uint32 // offset of the Slice in the Chunk
+	ID   uint64 // ID of the Slice, globally unique
+	Size uint32 // size of the Slice
+	Off  uint32 // offset of valid data in this Slice
+	Len  uint32 // size of valid data in this Slice
+}
+```
+
+This structure is encoded and saved in binary format, taking up 24 bytes.
+
+#### 3.1.9 SliceRef
+
+Records the reference count of a Slice, as follows
+
+```
+sliceId, size -> refs
+```
+
+Since the reference count of most Slice is 1, to reduce the number of related entries in the database, the actual value minus 1 is used as the stored count value in Redis and TKV. In this way, most of the Slices have a refs value of 0, and there is no need to create related entries in the database.
```

Comment:
```suggestion
Since the reference count of most Slices is 1, to reduce the number of related entries in the database, the actual value minus 1 is used as the stored count value in Redis and TKV. In this way, most of the Slices have a refs value of 0, and there is no need to create related entries in the database.
```